src/main/java/com/hms_networks/americas/sc/extensions/realtimedata/InstantValuesEbdString.java

```diff
@@ -0,0 +1,581 @@
+package com.hms_networks.americas.sc.extensions.realtimedata;
+
+import com.ewon.ewonitf.Exporter;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPoint;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPointBoolean;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPointNumber;
+import com.hms_networks.americas.sc.extensions.datapoint.DataPointString;
+import com.hms_networks.americas.sc.extensions.datapoint.DataQuality;
+import com.hms_networks.americas.sc.extensions.historicaldata.HistoricalDataManager;
+import com.hms_networks.americas.sc.extensions.string.StringUtils;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUnit;
+import com.hms_networks.americas.sc.extensions.system.time.SCTimeUtils;
+import com.hms_networks.americas.sc.extensions.taginfo.TagInfo;
+import com.hms_networks.americas.sc.extensions.taginfo.TagInfoManager;
+import com.hms_networks.americas.sc.extensions.taginfo.TagType;
+import com.hms_networks.americas.sc.extensions.util.StreamUtils;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class for working with the Ewon EBD instant values query using the string data output format.
+ * This class is used to parse the EBD data and create data points for each tag value. The string
+ * data output format for the instant values EBD includes tag data for all numeric and string tags.
+ *
+ * <p>It is important to note that the Ewon instant values EBD query does not return full tag
+ * configurations. The {@link TagInfoManager} is used to get tag information for each tag ID, and
+ * must be populated with tag information before using this class.
+ *
+ * <p>Format of the EBD data:
+ *
+ * <ul>
+ *   <li>The values of strings are surrounded by double quotes (“).
+ *   <li>Special characters (quote, double-quote, non-printable) are escaped with a ‘\’.
+ *   <li>Expected Header:
+ *       <ul>
+ *         <li>TagId
+ *         <li>TagName
+ *         <li>Value
+ *         <li>AlStatus
+ *         <li>AlType
+ *         <li>Quality
+ *       </ul>
+ *   <li>Followed by rows/lines for each data entry
+ * </ul>
+ *
+ * <p>Example:
+ *
+ * <pre>
+ * "TagId";"TagName";"Value";"AlStatus";"AlType";"Quality"
+ * 1;"Float_0";5;0;0;65472
+ * 2;"Float_1";6;0;0;65472
+ * 3;"Float_2";7;0;0;65472
+ * 2003;"Str_100";"ABCDEF\"GH\'IJKLMNOPQRSTUVWXYZ";0;0;65472
+ * 2004;"Str_101";"ABCDEFGHIJKLMNOPQRSTUVWXYZ";0;0;65472
+ * 2005;"Str_102";"AT char @";0;0;65472
+ * </pre>
+ *
+ * @author HMS Networks; Americas
+ * @version 1.0.0
+ * @since 1.15.15
+ */
+public class InstantValuesEbdString {
```

Comment:
Have you checked the performance of this vs the TagControl class? EBDs are very costly and it might be more performant to use TagControl even for large amounts of tags.

Reply:
Instant value EBD is roughly1100ms.  TagControl is 10ms per tag. EBD instant returns all tags. TagControl is each tag. The break even is around 110 tags. 
I think we can live with this performance, and improve the CPU utilization by implementing a non-CPU blocking wait for EBD response.